### PR TITLE
Adds integration hooks to getBoardTree()

### PR DESCRIPTION
**Pull request by Sesquipedalian**

### CODE REVIEW

File: Sources/Subs-Boards.php [hunk 1]
```diff
@@ -1307,18 +1307,25 @@ function getBoardTree()
 {
 	global $cat_tree, $boards, $boardList, $smcFunc;
 
+	$boardColumns = array('COALESCE(b.id_board, 0) AS id_board', 'b.id_parent', 'b.name AS board_name', 'b.description', 'b.child_level', 'b.board_order', 'b.count_posts', 'b.member_groups', 'b.id_theme', 'b.override_theme', 'b.id_profile', 'b.redirect', 'b.num_posts', 'b.num_topics', 'b.deny_member_groups', 'c.id_cat', 'c.name AS cat_name', 'c.description AS cat_desc', 'c.cat_order', 'c.can_collapse');
```
Comment (albertlast) [inline Sources/Subs-Boards.php]: some breakline would be improve the reading

Comment (Sesquipedalian) [reply Sources/Subs-Boards.php]: Agreed.

File: Sources/Subs-Boards.php [hunk 2]
```diff
@@ -1312,7 +1312,7 @@ function getBoardTree()
 		'b.description', 'b.child_level', 'b.board_order', 'b.count_posts', 'b.member_groups',
 		'b.id_theme', 'b.override_theme', 'b.id_profile', 'b.redirect', 'b.num_posts',
 		'b.num_topics', 'b.deny_member_groups', 'c.id_cat', 'c.name AS cat_name',
-		'c.description AScat_desc', 'c.cat_order', 'c.can_collapse'
+		'c.description AS cat_desc', 'c.cat_order', 'c.can_collapse',
```
Comment (albertlast) [inline Sources/Subs-Boards.php]: i believe that your create a invalid sql syntax if $extraBoardColumns is empty array 
specialy you implode on ,"space" and you added ,"newline"

Comment (Sesquipedalian) [reply Sources/Subs-Boards.php]: No. The newlines and commas here are merely separating the elements of the $boardColumns array. The newlines and commas are not part of any of the array elements themselves, so they will not be part of the imploded string.

Comment (Sesquipedalian) [reply Sources/Subs-Boards.php]: For a quick demo, run this code:
```
<?php
$boardColumns = array(
	'COALESCE(b.id_board, 0) AS id_board', 'b.id_parent', 'b.name AS board_name',
	'b.description', 'b.child_level', 'b.board_order', 'b.count_posts', 'b.member_groups',
	'b.id_theme', 'b.override_theme', 'b.id_profile', 'b.redirect', 'b.num_posts',
	'b.num_topics', 'b.deny_member_groups', 'c.id_cat', 'c.name AS cat_name',
	'c.description AS cat_desc', 'c.cat_order', 'c.can_collapse',
);
$extraBoardColumns = array();
$boardColumns = array_unique(array_merge($boardColumns, $extraBoardColumns));
echo implode(', ', $boardColumns);
```

The result is:
```
COALESCE(b.id_board, 0) AS id_board, b.id_parent, b.name AS board_name, b.description, b.child_level, b.board_order, b.count_posts, b.member_groups, b.id_theme, b.override_theme, b.id_profile, b.redirect, b.num_posts, b.num_topics, b.deny_member_groups, c.id_cat, c.name AS cat_name, c.description AS cat_desc, c.cat_order, c.can_collapse
```

Comment (Sesquipedalian) [reply Sources/Subs-Boards.php]: Has this addressed your concerns, @albertlast?

Comment (albertlast) [reply Sources/Subs-Boards.php]: sure

**Pull request merged**